### PR TITLE
[multistage] split out leaf stage operator

### PR DESCRIPTION
**Pull request by walterddr**
type canonicalization is not correct during split block; refactor the logic out to (1) LeafStageDataBlockOperator and (2) not doing implace replacement;

### CODE REVIEW

**Review by 61yao — APPROVED** [review]
LG

**Review by agavra — APPROVED** [review]
nice, I like this change a lot!

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/LeafStageTransferableBlockOperator.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import org.apache.pinot.core.operator.BaseOperator;
+import org.apache.pinot.core.operator.blocks.InstanceResponseBlock;
+import org.apache.pinot.core.operator.blocks.results.BaseResultsBlock;
+import org.apache.pinot.query.runtime.blocks.TransferableBlock;
+
+
+/**
+ * Leaf-stage transfer block operator is used to wrap around the leaf stage process results. They are passed to the
+ * Pinot server to execute query thus only one {@link DataTable} were returned. However, to conform with the
+ * intermediate stage operators. an additional {@link MetadataBlock} needs to be transfer after the data block.
```
Comment (snleee) [inline pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/LeafStageTransferableBlockOperator.java:42]: `operators. -> operators,` 😄

**Review by snleee — APPROVED** [review]

File: pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/LeafStageTransferableBlockOperator.java (hunk 1 above)
Comment (walterddr) [reply pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/LeafStageTransferableBlockOperator.java:42]:
```suggestion
 * intermediate stage operators. An additional {@link MetadataBlock} needs to be transferred after the data block.
```

**Pull request merged**